pages/resources/tutorials/query-history.md

```diff
@@ -0,0 +1,189 @@
+
+The content key is sent to the node as a parameter is a JSON object along with some metadata about the request. The basic form of a request is as follows:
+
+```json
+{
+    "jsonrpc":"2.0", "method": "<portal-method>", "params":[<content-key>], "id":1
+}
+```
+
+This can be sent to your node over HTTP using curl, as follows:
+
+```sh
+curl -X POST --data '{"jsonrpc":"2.0", "method": "portal_historyRecursiveFindContent", "params":["0x006f68209b1294f6365a7ef6c24d41a7195bc68831ae57261b277029196e00f6b0"], "id":1}'
+```
+
+## Checking available data
+
+Portal Network is being gradually filled up with historical data through bridge nodes. At the current time, not all of Ethereum's historical data has been uploaded into the network. This means that there is a reasonable chance that an arbitrary request will fail because the data is not yet available. The chance of arbitrary requests succeeding will increase as more data is uploaded. The more people run bridge nodes, the faster this process will be.
+
+To check what data has been uploaded to the network, you can use the network health monitoring tool, [Glados](http://glados.ethportal.net/content/).
+
+Glados allows you to view recent uploads into the network on the landing page. However, you can also append `key/<content-key>` to the landing page URL to lookup a specific content key to see if the data is currently available.
+
+Here's an example of a Glados entry for a recent successful upload:
+
+[Key 0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159](http://glados.ethportal.net/content/key/0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159/)
+
+You can tell from then key that this is a block header (begins `0x00` and the block hash was `0x06e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159` )
+
+
+So, to retrieve some real data from your Portal node, you can run the following command (or if you wish, replace the content key with another one that you have verified exists using Glados):
+
+```sh
+curl -X POST --data '{"jsonrpc":"2.0", "method": "portal_historyRecursiveFindContent", "params":["0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159"], "id":1}'
+```
+
+In the next stage you will want to access the data in the `content` field only, so you can amend your query to strip out the relevant data and dump it to a text file:
+
+```
+curl -X POST --data '{"jsonrpc":"2.0", "method": "portal_historyRecursiveFindContent", "params":["0x006e0ee92ebaf1b48919d193bd574b3096a85cc2609d286cdc511ad97ce5a18159"], "id":1}' | jq -r ".result.content" >> data.txt
+```
+
+
+## Decoding the response
```

Comment:
Nice, thanks for writing out these steps! For context, in `trin` we're mainly targeting using something like [web3.py](https://github.com/ethereum/web3.py) as the user interface. But at the current stage of development, your solution (a quick rust app) here is great and much needed for developers to get started interacting with portal data & becoming familiar with what's available.

It would be interesting to know if developers are finding this kind of tool/app useful (especially at devconnect). If there is demand we could definitely support something like this native to the `trin` codebase to make startup more efficient.